File: app/Models/Event.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Event extends Model
{
    protected $fillable = [
        'title',
        'description',
        'date',
        'location',
        'capacity',
    ];

    protected $hidden = ['updated_at'];

    // Formater la sortie JSON (snake_case → camelCase)
    protected function serializeDate(\DateTimeInterface $date): string
    {
        return $date->format('Y-m-d\TH:i:s\Z');
    }

    public function registrations()
    {
        return $this->hasMany(Registration::class);
    }

    // Nombre d'inscrits
    public function registrationsCount(): int
    {
        return $this->registrations()->count();
    }

    // Vérifie si l'événement est complet
    public function isFull(): bool
    {
        return $this->registrationsCount() >= $this->capacity;
    }
}

File: mobile/lib/models/event.dart
class Event {
  final int id;
  final String title;
  final String? description;
  final String date;
  final String location;
  final int capacity;
  final int registeredCount;
  final String createdAt;

  Event({
    required this.id,
    required this.title,
    this.description,
    required this.date,
    required this.location,
    required this.capacity,
    this.registeredCount = 0,
    required this.createdAt,
  });

  bool get isFull => registeredCount >= capacity;
  double get fillRate => registeredCount / capacity;

  factory Event.fromJson(Map<String, dynamic> json) => Event(
    id:              json['id'],
    title:           json['title'],
    description:     json['description'],
    date:            json['date'],
    location:        json['location'],
    capacity:        json['capacity'],
    registeredCount: json['registeredCount'] ?? 0,
    createdAt:       json['createdAt'] ?? '',
  );
}
